jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RuleContainer.java

```diff
@@ -185,7 +187,18 @@ protected String apply(String target, HttpServletRequest request, HttpServletRes
                     if (rule instanceof Rule.ApplyURI)
                         ((Rule.ApplyURI)rule).applyURI(baseRequest, baseRequest.getRequestURI(), encoded);
                     else
-                        baseRequest.setURIPathQuery(encoded);
+                    {
+                        String uriPathQuery = encoded;
+                        HttpURI baseUri = baseRequest.getHttpURI();
+                        // Copy path params from original URI if present
+                        if ((baseUri != null) && StringUtil.isNotBlank(baseUri.getParam()))
+                        {
+                            HttpURI uri = new HttpURI(uriPathQuery);
+                            uri.setParam(baseUri.getParam());
```

Comment:
We should probably note that our path param handling is fundamentally flawed in that we only handle parameters on the last segment of a path.... the spec allows `/foo;param1=value/bar;param2=value` but we don't handle that and I think the complexity of doing so in a rewrite scenario would be significant.

Reply:
I'm also wondering if we should limit this reinsertion of the parameters to only `jsesssionid`?
But if we do that, then we need to hook into the Session configuration stuff to know what that session keyname is.

Reply:
The closed issue #430 had some issues with this as well.
So I'm currently leaning towards not restricting it.

But in your example ... `/foo;param1=value/bar;param2=value`
The `HttpURI` parsing will parse that as the path = `/foo` and the param = `;param1=value/bar;param2=value` (ew!)